Added shared framework target to support Carthage.

regexident opened the pull request:
Follow-up for issue #19.

Carthage needs to find the xcodeproj and a shared framework target.
As such it is impedimental to have multiple xcodeproj files and subprojects, especially when the xcodeproj file has a different name than the Project and no shared framework.

regexident committed:
Added shared framework target to support Carthage.

regexident commented:
Any particular reason for closing this instead of merging it?

tomvanzummeren commented:
I forgot to explain. My apologies.

I had a whole restructuring going on including several pull requests that needed to be merged. So some pull requests (including this one) I simply did by hand on master because it was easier for me.

So if you look in master right now (and release 1.1.0), it should be Carthage compatible but I didn't test it. You would greatly help me out if you would be so kind to test it for me.

Thanks in advance!

tomvanzummeren commented:
Hmm.. now that I think of it, I only merged the Xcode projects, I did not add a shared framework target... If you would be so kind to add that as a pull request, I will merge it immediately!

regexident commented:
That's actually exactly what I did with this PR. Thus my question. :wink:

(It's based on commit 8f1e1b3, which as of right now if one commit behind and should merge without conflicts.)

What I did in this PR:
1. Renamed `TZStackView` demo app target to `TZStackViewDemo`.
2. Added `TZStackView` shared framework target.
3. Removed non-demo classes from demo target
4. Linked demo target with framework.

regexident committed:
Changed deployment target to iOS 8.

tomvanzummeren commented:
I am so sorry, there were lots of pull requests of weeks old, including one that rearranged all directories in order to support Carthage. I didn't realise yours wasn't even a day old and closed it because I confused it with that other one. I see I can cleanly merge this one so I am going to do it right now. Sorry for the confusion and thanks for your help!

tomvanzummeren commented:
Did it work? Does TZStackView now support Carthage?

regexident commented:
> I am so sorry, there were lots of pull requests of weeks old, including one that rearranged all directories in order to support Carthage. […] Sorry for the confusion and thanks for your help!

No problem! I figured it'd be a confusion of some kind.

> Did it work? Does TZStackView now support Carthage?

Looking good for me (as of aaef1db)! :+1: :heart_eyes:

To try it out yourself follow these simple steps:
1. [Install Carthage](https://github.com/Carthage/Carthage/releases), if you haven't already.
2. `$ mkdir test`.
3. `$ cd test`
4. `$ echo 'github "tomvanzummeren/TZStackView" "aaef1db"' > 'Cartfile'`
5. `$ carthage bootstrap`

You should get something like this:

```
*** No Cartfile.resolved found, updating dependencies
*** Fetching TZStackView
*** Checking out TZStackView at "aaef1db47cedeebdbfcc3b6b23412924a2500e7f"
*** xcodebuild output can be found in /var/folders/pc/2nfs9w4n2p5gr4r22cqnt5jcsh_nv5/T/carthage-xcodebuild.YVa7uO.log
*** Building scheme "TZStackView" in TZStackView.xcodeproj
```

…and find a directory `Carthage` added to `test` with subdirectories `Build` and `Checkout` containing your artifacts.

Thanks a lot for your work, Tom! We owe you one. :beer:

tomvanzummeren commented:
You're welcome! And thank YOU!
